pkg/controller/ec2/vpccidrblock/controller.go

```diff
@@ -0,0 +1,195 @@
+		return managed.ExternalObservation{}, errors.New(errMultipleItems)
+	}
+
+	observed := response.Vpcs[0]
+
+	currentStatusCode, err := ec2.FindVPCCIDRBlockStatus(meta.GetExternalName(cr), observed)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errDescribe)
+	}
+
+	switch currentStatusCode { //nolint:exhaustive
+	case awsec2.VpcCidrBlockStateCodeAssociated:
+		cr.SetConditions(xpv1.Available())
+	case awsec2.VpcCidrBlockStateCodeAssociating:
+		cr.SetConditions(xpv1.Creating())
+	case awsec2.VpcCidrBlockStateCodeDisassociated:
+		cr.Status.SetConditions(xpv1.Deleting())
+	case awsec2.VpcCidrBlockStateCodeDisassociating:
+		cr.Status.SetConditions(xpv1.Deleting())
+	default:
+		cr.SetConditions(xpv1.Unavailable())
+	}
+
+	cr.Status.AtProvider = ec2.GenerateVpcCIDRBlockObservation(meta.GetExternalName(cr), observed)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ec2.IsCIDRNotFound, err), errDescribe)
+	}
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: true,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.VPCCIDRBlock)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	result, err := e.client.AssociateVpcCidrBlockRequest(&awsec2.AssociateVpcCidrBlockInput{
+		AmazonProvidedIpv6CidrBlock:     cr.Spec.ForProvider.AmazonProvidedIPv6CIDRBlock,
+		CidrBlock:                       cr.Spec.ForProvider.CIDRBlock,
+		Ipv6CidrBlock:                   cr.Spec.ForProvider.IPv6CIDRBlock,
+		Ipv6CidrBlockNetworkBorderGroup: cr.Spec.ForProvider.IPv6CIDRBlockNetworkBorderGroup,
+		Ipv6Pool:                        cr.Spec.ForProvider.IPv6Pool,
+		VpcId:                           cr.Spec.ForProvider.VPCID,
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, awsclient.Wrap(err, errAssociate)
+	}
+
+	if result != nil {
+		if result.CidrBlockAssociation != nil {
+			meta.SetExternalName(cr, aws.StringValue(result.CidrBlockAssociation.AssociationId))
+		}
+		if result.Ipv6CidrBlockAssociation != nil {
+			meta.SetExternalName(cr, aws.StringValue(result.Ipv6CidrBlockAssociation.AssociationId))
+		}
+	}
+
+	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+}
+
+func (e *external) Update(_ context.Context, _ resource.Managed) (managed.ExternalUpdate, error) {
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.VPCCIDRBlock)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+	if ec2.IsVpcCidrDeleting(cr.Status.AtProvider) {
+		return nil
+	}
+
+	_, err := e.client.DisassociateVpcCidrBlockRequest(&awsec2.DisassociateVpcCidrBlockInput{
+		AssociationId: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	return awsclient.Wrap(resource.Ignore(ec2.IsVPCNotFoundErr, err), errDisassociate)
```

Comment:
This is actually a valid error for this resource. If it exists, we should ignore `NotAssociated`